service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/SpecimenMapper.java

```diff
@@ -28,69 +35,198 @@
+    public String mapSpecimenToCompoundStatement(Specimen specimen, List<Observation> observations, String diagnosticReportIssuedDate) {
+        String mappedObservations = mapObservationsAssociatedWithSpecimen(specimen, observations);
+
+        var specimenCompoundStatementTemplateParameters = SpecimenCompoundStatementTemplateParameters.builder()
+            .compoundStatementId(messageContext.getIdMapper().getOrNew(ResourceType.Specimen, specimen.getId()))
+            .diagnosticReportIssuedDate(diagnosticReportIssuedDate)
+            .effectiveTime(prepareEffectiveTimeForSpecimen(specimen))
+            .observations(mappedObservations);
+
+        buildType(specimen).ifPresent(specimenCompoundStatementTemplateParameters::type);
+        buildAccessionIdentifier(specimen).ifPresent(specimenCompoundStatementTemplateParameters::accessionIdentifier);
+        buildPertinentInformation(specimen).ifPresent(specimenCompoundStatementTemplateParameters::pertinentInformation);
+        buildParticipant(specimen).ifPresent(specimenCompoundStatementTemplateParameters::participant);
+
+        return TemplateUtils.fillTemplate(
+            SPECIMEN_COMPOUND_STATEMENT_TEMPLATE,
+            specimenCompoundStatementTemplateParameters.build()
+        );
+    }
+
+    public static String prepareEffectiveTimeForSpecimen(Specimen specimen) {
+        Optional<DateTimeType> effectiveDate = getCollectionDateTime(specimen);
+
+        return effectiveDate.map(date -> String.format(EFFECTIVE_TIME_CENTER_TEMPLATE, DateFormatUtil.toHl7Format(date)))
+            .orElse(DEFAULT_TIME_VALUE);
+    }
+
+    private static Optional<DateTimeType> getCollectionDateTime(Specimen specimen) {
+        Optional<DateTimeType> effectiveDate = Optional.empty();
+
+        if (specimen.hasReceivedTime()) {
+            effectiveDate = Optional.of(specimen.getReceivedTimeElement());
+        }
+
+        if (specimen.hasCollection()) {
+            Specimen.SpecimenCollectionComponent collection = specimen.getCollection();
+            if (collection.hasCollectedDateTimeType()) {
+                effectiveDate = Optional.of(collection.getCollectedDateTimeType());
+            } else if (collection.hasCollectedPeriod()) {
+                effectiveDate = Optional.of(collection.getCollectedPeriod().getStartElement());
+            }
+        }
+
+        return effectiveDate;
+    }
+
+    private Optional<String> buildParticipant(Specimen specimen) {
+        if (specimen.hasCollection() && specimen.getCollection().hasCollector()) {
+            Reference collector = specimen.getCollection().getCollector();
+
+            return Optional.of(messageContext.getIdMapper().get(collector));
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<String> buildAccessionIdentifier(Specimen specimen) {
+        if (specimen.hasAccessionIdentifier()) {
```

Comment:
should we check if the accession identifier has a value too?

```suggestion
        if (specimen.hasAccessionIdentifier() && specimen.getAccessionIdentifier().hasValue()) {
```